unittests/test_schedulers.py

```diff
@@ -525,16 +525,25 @@ def test_cancel_with_grace(minimal_job, scheduler, local_only):
     # signal handler for SIGTERM
     time.sleep(1)
 
+    # Try reading the pid of spawned sleep, until a valid value is retrieved
+    for i in range(3):
+        try:
+            with open(minimal_job.stdout) as fp:
+                sleep_pid = int(fp.read())
+        except ValueError:
+            time.sleep(1)
+            continue
+        else:
+            break
```

Comment:
I think we don't need the `else` here since you've got the `continue` when you catch the exception.
```suggestion

        break
```

Reply:
I need it in order to break out of the loop without raising the pytest failure in case the pid is retrieved

Reply:
Sure, but you'd still be breaking out of the loop without the `else`. If you move the `break` statement after the `try`/`except` block, it will only be executed when the `try` succeeds.

Alternatively, you can leave the `break` in that `else`, but then I'd suggest to remove the `continue` from the `except` since it is redundant here.

Reply:
Ok, it's clearer now, thanks.